### PR TITLE
NodeJS Chat Example

### PR DESCRIPTION
**Pull request by joeyguerra**
It wasn't easy for me to figure out how to use the chat module in a NodeJS application for a bot. I was just thinking that if this example that I added was here already, it would've saved me a few hours.

Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ x ] Code coverage does not decrease (if any source code was changed)
- [ x ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by joeyguerra** [comment]
I signed it!

File: src/apis/chat/README.md [hunk 1]
```diff
@@ -33,6 +33,30 @@ const { chat, auth } = Chat;
 </script>
 ```
 
+## Quick NodeJS example
```
Comment (bcoe) [inline src/apis/chat/README.md]: Just to be consistent with a bunch of our other libraries (we use a template on modules such as `nodejs-storage`) let's make this heading:

`## Using the client library`

File: src/apis/chat/README.md [hunk 2]
```diff
@@ -33,6 +33,30 @@ const { chat, auth } = Chat;
...
+
+```js
+// Assuming that `GOOGLE_APPLICATION_CREDENTIALS="/app/.data/chatbot.json"` is set as an environment variable with the path to the json key that you downloaded when creating a service account for your bot.
+
+const {google} = require("googleapis")
+const SCOPES = ["https://www.googleapis.com/auth/chat.bot"]
+
+const YourModule = {
```
Comment (bcoe) [inline src/apis/chat/README.md]: I'd be tempted to drop the `YourModule` bit, and just have:

```js
  async main(){
    const auth = await google.auth.getClient({
      scopes: SCOPES
    })
    const chat = google.chat({version: "v1", auth: auth})
    console.log(chat)
    const project = await google.auth.getProjectId()
    const spaces = await chat.spaces.list()
  }
}

mainj()
```

:+1:

**Comment by joeyguerra** [comment]
I've incorporated @bcoe 's suggestions. Any guidance on what to do about the other required checks?

**Comment by bcoe** [comment]
@joeyguerra thanks a ton for the contribution \o/

**Comment by joeyguerra** [comment]
Awesome! My pleasure. Thank you for such a great experience. This is well done.

**Comment by JustinBeckwith** [comment]
Thanks for the PR @joeyguerra!

**Pull request merged**